python/cudf/cudf/core/multiindex.py

```diff
@@ -256,7 +256,7 @@ def astype(self, dtype, copy: bool = True):
         return self
 
     @_performance_tracking
-    def rename(self, names, inplace=False):
+    def rename(self, names, inplace: bool = False) -> Self | None:
```

Comment:
1. Does Pandas have its own version of `Self` that it uses for doing type annotations?
2. Do you actively look for where we're missing type annotations, or it more a "you fix as you see it" thing?

Reply:
1. pandas uses the same `Self` from the `typing_extensions` package https://github.com/pandas-dev/pandas/blob/main/pandas/_typing.py#L102-L107
2. More of the latter, "fix as I see it". This PR is somewhat of an exception. Generally if I modify any part of cudf Python I also try to add "easy" typing annotations around the modification.

Reply:
Ah okay, thanks!